cstar/tests/roms/fixtures.py

```diff
@@ -0,0 +1,155 @@
+        pup_test_data = pooch.create(
+            path=ROMS_TOOLS_DATA_DIRECTORY,
+            base_url="https://github.com/CWorthy-ocean/roms-tools-test-data/raw/main/",
+            # The registry specifies the files that can be fetched
+            registry={
+                "GLORYS_NA_2012.nc": "b862add892f5d6e0d670c8f7fa698f4af5290ac87077ca812a6795e120d0ca8c",
+                "ERA5_NA_2012.nc": "d07fa7450869dfd3aec54411777a5f7de3cb3ec21492eec36f4980e220c51757",
+                "TPXO_global_test_data.nc": "457bfe87a7b247ec6e04e3c7d3e741ccf223020c41593f8ae33a14f2b5255e60",
+                "CESM_BGC_2012.nc": "e374d5df3c1be742d564fd26fd861c2d40af73be50a432c51d258171d5638eb6",
+                "CESM_BGC_SURFACE_2012.nc": "3c4d156adca97909d0fac36bf50b99583ab37d8020d7a3e8511e92abf2331b38",
+            },
+        )
+        pup_test_data.fetch("GLORYS_NA_2012.nc")
+        pup_test_data.fetch("TPXO_global_test_data.nc")
+        pup_test_data.fetch("ERA5_NA_2012.nc")
+        pup_test_data.fetch("CESM_BGC_2012.nc")
+        pup_test_data.fetch("CESM_BGC_SURFACE_2012.nc")
+
+        if symlink_path.is_symlink():
+            raise FileExistsError(
+                f"{symlink_path} already exists as a symlink to {symlink_path.resolve()}"
+            )
+
+        symlink_path.symlink_to(ROMS_TOOLS_DATA_DIRECTORY)
+
+        def cleanup():
+            if symlink_path.is_symlink():
+                print(f"removing {symlink_path}")
+                symlink_path.unlink()
+
+        request.addfinalizer(cleanup)
+
+    return _fetch_roms_tools_source_data
+
+
+@pytest.fixture
+def fetch_remote_test_case_data() -> Callable[[], None]:
+    """
+    Fixture that provides a function to fetch remote test case data from a GitHub repository.
+
+    Returns
+    -------
+    Callable[[], None]
+        A factory function that fetches the remote test case data and sets it up in the local
+        test data directory.
+    """
+
+    def _fetch_remote_test_case_data() -> None:
+        """
+        Downloads and sets up the remote test case data for testing.
+
+        This function downloads a zip archive of a specific commit of a repo containing test data,
+        extracts the archive into the specified test data directory, and performs cleanup of
+        intermediate files and directories.
+
+        Data are saved to the CSTAR_TEST_DATA_DIRECTORY, set in tests/config.py
+
+        Returns
+        -------
+        None
+        """
+
+        test_case_repo_url = (
+            "https://github.com/CWorthy-ocean/cstar_blueprint_test_case/"
+        )
+        checkout_target = "d7996aea8d4fd4ca4148b34d2d898f019c90a8ff"
+
+        # Construct the URL of this commit as a zip archive:
+        archive_url = f"{test_case_repo_url.rstrip('/')}/archive/{checkout_target}.zip"
+
+        # Create a cache dir with pooch
+        # cache_dir = Path(pooch.os_cache("cstar_test_case_data"))
+
```

Comment:
delete

Reply:
Still needs to be deleted!

Reply:
done in 7ed3a60c751d0c36614db2ff2658918f2d682cb0